src/commands/basemaps-github/github.ts

```diff
@@ -2,115 +2,220 @@ import { Env, LogType } from '@basemaps/shared';
    */
-  commit(message: string): void {
-    this.logger.info({ repository: this.repo }, 'GitHub: Commit all');
-    execFileSync('git', ['commit', '-am', message], { cwd: this.repoName }).toString().trim();
+  async createCommit(blobs: Blob[], message: string, sha: string, logger: LogType): Promise<string> {
+    // Create a tree which defines the folder structure
+    logger.debug({ sha }, 'GitHub API: Create Tree');
+    const treeRes = await this.octokit.rest.git.createTree({
+      owner: this.owner,
+      repo: this.repo,
+      base_tree: sha,
+      tree: blobs,
+    });
+    if (!this.isOk(treeRes.status)) throw new Error(`Failed to create tree.`);
+
+    const treeSha = treeRes.data.sha;
+
+    // Create the commit
+    logger.debug({ treeSha }, 'GitHub API: Create Commit');
+    const commitRes = await this.octokit.rest.git.createCommit({
+      owner: this.owner,
+      repo: this.repo,
+      message,
+      parents: [sha],
+      tree: treeSha,
+    });
+    if (!this.isOk(commitRes.status)) throw new Error(`Failed to create commit.`);
+    return commitRes.data.sha;
   }
 
   /**
-   * Push the local brach
-   *
+   * Update the reference of your branch to point to the new commit SHA
    */
-  push(): void {
-    this.logger.info({ repository: this.repo }, 'GitHub: Push');
-    execFileSync('git', ['push', 'origin', 'HEAD'], { cwd: this.repoName }).toString().trim();
+  async updateBranch(branch: string, commitSha: string, logger: LogType): Promise<void> {
+    logger.debug({ branch, commitSha }, 'GitHub API: update ref');
+    const response = await this.octokit.rest.git.updateRef({
+      owner: this.owner,
+      repo: this.repo,
+      ref: this.toRef(branch),
+      sha: commitSha,
+    });
+    if (!this.isOk(response.status)) throw new Error(`Failed to update branch ${branch} sha.`);
   }
 
   /**
-   * Create pull request
-   * This needs to use github API to create Pull request by access token.
-   *
+   * Create a new pull request from the given branch and return pull request number
    */
-  async createPullRequests(branch: string, title: string, draft: boolean): Promise<number> {
+  async createPullRequest(branch: string, title: string, logger: LogType): Promise<number> {
     // Create pull request from the give head
     const response = await this.octokit.rest.pulls.create({
-      owner: this.org,
-      repo: this.repoName,
+      owner: this.owner,
+      repo: this.repo,
       title,
       head: branch,
       base: 'master',
-      draft,
     });
     if (!this.isOk(response.status)) throw new Error('Failed to create pull request.');
-    this.logger.info({ branch, url: response.data.html_url }, 'GitHub: Create Pull Request');
+    logger.info({ branch, url: response.data.html_url }, 'GitHub: Create Pull Request');
     return response.data.number;
   }
 }
+
+export interface GithubFiles {
+  path: string;
+  content: string;
+}
+
+export class GitHubCreatePR {
```

Comment:
This doesnt need to be a class, could just make it a function, saves you doing `new GithubFileS(xxx).createPR()`